cmake/gcc_dev_tools.cmake

```diff
@@ -0,0 +1,30 @@
+# Run a set of GCC tools on the code base to maintain high code standards
+# and catch potential security vulnerabilities
+#
+# Tools ran:
+# - cppcheck
+
+option(
+  USE_CPP_CHECK
+  "Use cppcheck if the program is found."
+  OFF
+)
+
+# Adding clang-format target if executable is found
+find_program(CPP_CHECK "cppcheck")
+if(CPP_CHECK)
+  message("cppcheck found!")
+  if(USE_CPP_CHECK)
```

Comment:
Similarly here, let's put the `USE_CPP_CHECK` condition at the top level (so we don't bother checking for the tool if it won't b used).  If the tool is not found, it can be an error, to catch mistakes.

Also, since we are printing a message when the tool is enabled and found it could be helpful to print `To run, use: make cppcheck`, so it's 100% clear.